lib/notifications/notification.ex

```diff
@@ -141,26 +201,17 @@ defmodule Notifications.Notification do
 
   @spec unexpired_notifications_for_user(DbUser.id(), (-> Util.Time.timestamp())) :: [t()]
   def unexpired_notifications_for_user(user_id, now_fn \\ &Util.Time.now/0) do
-    cutoff_time = now_fn.() - @notification_expiration_threshold
+    import Notifications.Db.Notification.Queries
 
-    query =
-      from(n in DbNotification,
-        join: nu in assoc(n, :notification_users),
-        join: u in assoc(nu, :user),
-        left_join: bw in assoc(n, :block_waiver),
-        left_join: bm in assoc(n, :bridge_movement),
-        select: %DbNotification{
-          id: n.id,
-          created_at: n.created_at,
-          state: nu.state,
-          block_waiver: bw,
-          bridge_movement: bm
-        },
-        where: n.created_at > ^cutoff_time and u.id == ^user_id,
-        order_by: [desc: n.created_at]
-      )
+    cutoff_time = now_fn.() - @notification_expiration_threshold
 
-    query
+    base()
+    |> select_user_read_state(user_id)
+    |> select_bridge_movements()
+    |> select_block_waivers()
+    |> select_detour_info()
+    |> where([notification: n], n.created_at > ^cutoff_time)
```

Comment:
hmm, gonna need to figure out a better way to do this now... since detours can be active for longer than the cutoff time..

Reply:
Do we even need/want to limit detour lifespan? I figured that a detour would just be active until it's deactivated.

Reply:
You're correct, but block waivers and bridge movements don't agree and do require this, so I need a way to only apply this for non detour notifications